contract/AElf.Contracts.Profit/ProfitContract.cs

```diff
@@ -229,350 +283,481 @@ public override Empty SubWeight(SubWeightInput input)
+                $"Available details: {availableDetails.Aggregate("\n", (profit1, profit2) => profit1.ToString() + "\n" + profit2.ToString())}");
 
-            for (var i = 0; i < Math.Min(ProfitContractConsts.ProfitLimit, availableDetails.Count); i++)
+            // Only can get profit from last profit period to actual last period (profit.CurrentPeriod - 1),
+            // because current period not released yet.
+            for (var i = 0;
+                i < Math.Min(ProfitContractConsts.ProfitReceivingLimitForEachTime, availableDetails.Count);
+                i++)
             {
                 var profitDetail = availableDetails[i];
                 if (profitDetail.LastProfitPeriod == 0)
                 {
+                    // This detail never performed profit before.
                     profitDetail.LastProfitPeriod = profitDetail.StartPeriod;
                 }
 
-                var lastProfitPeriod = profitDetail.LastProfitPeriod;
-                //Can only get profit until profitItem.CurrentPeriod-1,because currentPeriod hasn't be released.
-                for (var period = profitDetail.LastProfitPeriod;
-                    period <= (profitDetail.EndPeriod == long.MaxValue
-                        ? profitItem.CurrentPeriod - 1
-                        : Math.Min(profitItem.CurrentPeriod - 1, profitDetail.EndPeriod));
-                    period++)
+                ProfitAllPeriods(profitItem, input.Symbol, profitDetail, profitVirtualAddress);
+            }
+
+            State.ProfitDetailsMap[input.SchemeId][Context.Sender] = new ProfitDetails {Details = {availableDetails}};
+
+            return new Empty();
+        }
+
+        private long ProfitAllPeriods(Scheme scheme, string symbol, ProfitDetail profitDetail,
+            Address profitVirtualAddress, bool isView = false)
+        {
+            var totalAmount = 0L;
+            var lastProfitPeriod = profitDetail.LastProfitPeriod;
+            for (var period = profitDetail.LastProfitPeriod;
+                period <= (profitDetail.EndPeriod == long.MaxValue
+                    ? scheme.CurrentPeriod - 1
+                    : Math.Min(scheme.CurrentPeriod - 1, profitDetail.EndPeriod));
+                period++)
+            {
+                var periodToPrint = period;
+                var detailToPrint = profitDetail;
+                var releasedProfitsVirtualAddress =
+                    GetReleasedPeriodProfitsVirtualAddress(profitVirtualAddress, period);
+                var releasedProfitsInformation = State.DistributedProfitsMap[releasedProfitsVirtualAddress];
+                if (releasedProfitsInformation == null || releasedProfitsInformation.TotalShares == 0)
+                {
+                    continue;
+                }
+
+                Context.LogDebug(() => $"Released profit information: {releasedProfitsInformation}");
+                var amount = profitDetail.Shares.Mul(releasedProfitsInformation.ProfitsAmount[symbol])
```

Comment:
`.Mul` may overflow if the two operands are too big and the method cannot be successful. One option to solve this is to convert to `decimal` to do calculation and then convert back.